### DISCUSSION TITLE
Ideas beyond 1.0

### DISCUSSION THREAD

**Discussion by Rearth** [Rearth/Oritech#20] [Ideas]
Here's a short list of some ideas that won't make it into the 1.0 release, but might be added at a later point. Everyone feel free to add suggestions.

- [x] Particle Accelerator, for producing lategame materials
- [ ] Nuclear / Fission Reactors. These will be multiblocks, maybe similar in design to big reactors? Possible with more flexibility in arrangement, as I do love the old ic2 reactor design challenges
- [ ] Pipe animations. They transport things instantly, but we could fake it at the client by showing a fast item moving through it. With energy we could just show the entire network pulsating.
- [ ] Sentry towers, guarding an area around them and shooting at mobs
- [ ] Forcefields?
- [ ] Void energy access / channelling / transformation machines
- [ ] A transport / combat drone buddy that follows the player. Can be outfitted with different modules, such as chest, gun, ore finder, etc.
- [ ] Teleporters
- [ ] Spatial transportation device - box built around an area, which will drop it as item (for spawners or base relocation)
- [ ] New (electric/proimetheum) bow types

**Comment by JasdanVM** [comment]
Advanced Weapons
Starting from a trash launcher and non-gunpowder simple guns all the way to
Lasers, Rail-guns, electromagnetic cannons, mechanical fists, chainsaws, vibroblades, plasma rifles, shock-electric pistols, etc.

**Comment by jshipley** [comment]
It would be fun to add a few security related blocks for the Industrial door (but also usable for other things). They could just emit a redstone pulse when the right key/entity is used to unlock it.

* Keycard / card slot
* Player detector (bonus points if there's a scan animation when living entities are close enough)

As for turrets, a great start would be giving the Enderic Lasers a defense mode (through an upgrade?) that makes them target hostile mobs. Or let them attack any living entities that are not whitelisted (for multiplayer fun).

**Comment by pourgb** [comment]
I would love a reactor, producing power is too hard at this point, I have like 10 solar panels and they just cant keep up with all of my machines👀

**Reply by Rearth** [reply]
Yeah a reactor is definitly relativly high on my next feature list. However there's already plenty of good power options available. A fuel generator with some steam boilers and upgrades can already work wonders, and can easily be scaled

**Reply by JasdanVM** [reply]
Isn't a reactor just a more fancy way to boil water tho?

**Reply by Rearth** [reply]
Well yes. The existing steam engines would be utilized with the reactor. I'll probably offer 2 operation modes, one that directly outputs energy, and one that outputs steam. I think big reactors did that too

**Reply by viro233** [reply]
你应该给你的机械加上升级，让它更快耗电更少，或者给你的发电加上升级，让它发更多的电且使用更少的资源。我在加上57个速度后加上效率升级，机械的运行速度达到了1t/s，能量消耗只有几fe，当然这仍然不是机械的极限，这是一个很强的模组，给熔岩发电机加点升级，你甚至能超过mekanism的聚变反应堆，消耗的仅有原版就可以无限的熔岩，而且消耗的速率可以降的很低
![Uploading 7414f025120ee0aa798a5c6902a9d593.png…]()
![42d3d82041d44c8265c61dd743c3f526](https://github.com/user-attachments/assets/fd451b60-02a9-41ff-b805-eca4dbc487cb)
![0dd6d9b3a75b53da3ac3db29cf05ff68](https://github.com/user-attachments/assets/a1ed4630-54f1-489c-b4bb-3797cf4bc422)
我觉得应该加强蒸汽发电，而不是让蒸汽发电跟太阳能在一个水平线上
![a1458c112ab17720871536e6e7b1a9b5](https://github.com/user-attachments/assets/b2f11d41-c787-4b33-9e22-db63440fee5e)
![c293a7eb49b34f192b5970e69f8f33be](https://github.com/user-attachments/assets/4ae538b2-14d2-4b5d-bcd1-a3455145daad)

**Reply by 0wdg** [reply]
speechless at the images

**Comment by CHN-YunRu** [comment]
- Like the Mekanism Generators, fission reactors can be used to generate steam. Fusion reactors can generate electricity directly, and when coolant is passed through, steam is generated. Steam can be passed into turbines to generate electricity. It is also possible to add more ways to obtain steam on the existing basis. It would be exciting if these ideas were realized in the form of this multi-block structure.
- For weapons, i think you can refer to the EMX-Arms gunpack in Forge Mod-Timeless and Classics Zero. Railguns, laser guns, and various cold weapons using future technology are all very interesting ideas. We can even use various blocks to form a large electromagnetic cannon likes in Create Big Cannons, which is also in line with the characteristics of this mod.

**Comment by jshipley** [comment]
For guns/bows/etc. I think that a handheld enderic laser could be really nice. You could use it to mine or kill mobs, and it could reuse some of the code (behavior, not searching/targeting) from the enderic laser classes.

It could mine/attack from a distance (and send the drops straight into your inventory?) It should be enchantable with efficiency/fortune/looting/power/etc. The battery in it shouldn't last for too long, though. It would also make the exo chestplate continue being useful even after you get a prometheum pickaxe.

**Comment by MinejekHD** [comment]
A hydroelectric power plant (like a dam) or a hydrogen power plant would also be cool ideas. The production of hydrogen in particular would be really interesting and unique for this mod. 
Or different types of combustion engines (4 or 6 cylinder engines) which, depending on their size, provide more power

**Reply by EmmaTheMartian** [reply]
The only mod I've ever seen hydroelectric power in was Emerging Technology, which unfortunately is unmaintained and only on 1.12.2. I'd love to see hydroelectric power again

**Comment by GoldenMerc** [comment]
Could you add raw ore blocks in the next update so mining nickel is easier?

**Reply by Rearth** [reply]
Good news: That was just added to the code a few days ago and will be included in the next release

**Comment by Rislir** [comment]
Hi! A cool mod, a new breath among industrial mods. 
When are you planning to add a nuclear reactor to the mod? I agree that he was interesting enough in IC2. It's especially cool that it could overheat or explode from the thoughtless arrangement of components. 
And I also wonder when the guide book will be inside the game?

**Comment by OneHwonder** [comment]
Idea for a change to the chiller. Instead of having it as a basically freezer, you could call it a crystalizer. it basically grows crystals or turns materials into their crystal form. this kinda keeps the original recipes of turning water into ice and steam into snow (snowflakes count as crystals I'm pretty sure). you could add more recipes like: 
Lava into obsidian 
magma into basalt (surprisingly a type of crystal)
Amethyst shard into Amethyst cluster (allowing for lab-grown crystals!)
uranium into uranite crystal
Nether quartz into quartz block (or quartz cluster like with amethyst, but you'd need to add that as well)

Obviously these recipes basically duplicate items so you'd make it take upwards of 5+ minutes to finish the recipe (the time could change depending on the recipe)

You could even add some fantasy recipes like:
glowstone into glowstone block (it kinda looks like a crystal)
Prismarine crystal into prismarine lamp
Ice into packed ice 
Packed ice into blue ice (the ice recipes assume that you're basically just growing more ice crystals on top of the ice block making it thicker? idk)
faster oxidization of copper blocks (rusting is not the same as crystalization so it could be considered a stretch to implement)
Sand into glass (glass is not a crystal but if you suspend your disbelief a little it kinda makes sense)
or even honey combs into honey comb blocks (that's just way out there but combs are repeating patterns, kinda like a crystal. except bees make them so idk)

**Comment by Verifiable** [comment]
The mod doesn't currently provide a way to distribute items among multiple outputs, making things like automating the Foundry's 2 inputs much more difficult than it could be. You might think the inventory proxy addon could be used for this purpose, but since pipes always output to the ambiguously-closest inventory, one slot will always remain empty until the first slot fills up (pictured) 
![image](https://github.com/user-attachments/assets/19b36cc1-42c1-4b57-aef0-8215f0f0e5dc)

Similarly, pipes in extraction mode only drawing from the first non-empty inventory slot makes it unnecessarily easy to "jam" a network, requiring you to set up a "handler" network for each and every item type you expect to appear, lest that item ends up in the first slot and blocks further extractions. Unless it's performance-related, I can't really see any reason why they should be limited in this way.

I'd like to suggest that a daisy-chainable "distributor block" be added to ease material processing stage automation, and the removal of the first-non-empty-slot limitation on extraction pipes.

I'd also like to suggest extraction pipes being made compatible with the vanilla Crafter block - currently, they'll draw from the Crafter's inputs and ignore the outputs: the Crafter will just spit the crafted item out into the world. You can overcome this by adding a Hopper on the output of the Crafter, with the extraction pipe connected to that, but it seems a bit unnecessary.

**Reply by LaCumbia** [reply]
Second this heavy. Also the fact that you can't pass items through item filters.

**Comment by LaCumbia** [comment]
This a complaint about a very _very_ minor detail. There is a piece of math for the nuclear reactor.
The amount of heat generated per tick

`pulses / 2 * pulses + 4`

could be written as

`( pulses ** 2 / 2 ) + 4`

Now I have 0 idea as if that would optimize anything but just mathematically it's more compact.

**Comment by ciefa** [comment]
Big and in-depth reactors for power would be amazing.

**Comment by MUZAKEYSAR** [comment]
Hello! I hope I'm not being too demanding or bothersome, but I absolutely love the mod and have a few suggestions that I think could be great additions for the future.

Tools
Handheld Sonar: An item held in either hand that detects entities in a 30-block radius. It could be upgraded to allow you to create a whitelist/blacklist, so you can filter which specific entities you want it to highlight.

Grappling Hook: Allows you to propel yourself to the point you anchor it to. It could have several upgrade paths:

Multi-Hook: Lets you fire more than one cable, allowing you to hang between two or more points. This would be amazing for building.

Tether Hook: Allows you to anchor it to entities, either pulling them towards you or pulling you towards them.

3x3 Hammer: A hammer that mines a 3x3 area, but only of a single block type (e.g., it would break all the Stone in an area but leave any Ores untouched). This would be incredibly useful for clearing out large areas without destroying valuables.

Structure Projector: A tool that can copy a structure within a 10x10x10 area and then display a hologram of it wherever you aim. Upgrades could include a larger copy area or even an "auto-build" feature that constructs the hologram for you if you have the required materials in your inventory.

Modular Vehicles
Instead of multiple separate machines, it would be amazing to have a single "base chassis" that can be modified for different situations:

Add Wheels to turn it into a land rover (faster than a horse, but can only climb one block high).

Add Pontoons to turn it into a fast boat.

Add Thrusters/Rotors to turn it into a flying machine (slower than an Elytra, but much easier to maneuver and capable of hovering).

Add an Oxygen System and Glass Canopy to turn it into a submarine.

This modular vehicle could also have its own upgrade slots for things like: extra storage, a sonar system to detect entities, a Drill Head for mining, a passenger seat, and even mountable turrets for combat.

Drones & Turrets
Drones: In addition to companion drones, it would be cool to have Guard Drones. You could assign them a patrol perimeter, turning them into mobile turrets. To keep them balanced, hostile mobs should be able to damage them. If they are destroyed, there could be a mechanic to repair them. They would recharge at a "Charging Station" placed within their patrol area. You could also add utility upgrades to give them other tasks, like chopping wood or fishing.

Turrets: Similar to the vehicle idea, turrets could be a modular base. You would then add different weapon modules to change their function, such as:

Standard Projectiles

Incendiary (Fire) or Cryo (Freezing) ammunition

TNT Launchers

Ender-Energy Beams

An awesome upgrade would be the ability to manually control the turret for better immersion and fun.

Tech Golem
This could be an upgrade path for the vanilla Iron Golem. You could give them specific orders and improve them in various ways. They could have their own charging stations and even a mechanic where they can mount and operate a nearby turret if it's within their assigned perimeter. Upgrades could include damage resistance, ranged attacks, etc.

The Engineer Villager
A new villager profession created with a specific workstation. The Engineer's role, besides trading mod-related items, would be to automatically repair nearby drones, turrets, and Tech Golems. This would be a great way to automate the maintenance of your base.

Modular Armor
It would be incredible to have an "Iron Man-style" armor station that lets you install upgrades into special slots on the mod's armor pieces. The key would be a limited number of slots per piece, forcing players to make strategic choices. Upgrade ideas include:

Oxygen Tanks: Refillable tanks to breathe underwater for longer.

Propulsion Fins: For faster movement in water.

Magnetic Plating: To reduce knockback.

Effect Filters: To reduce the duration of negative effects like poison or slowness.

Mechanical Elytra: Works with energy instead of fireworks.

Arm-Mounted Projectile Launchers.

Specialized Plating: To reduce specific damage types (fire, blast, projectile).

Force Field Core: A module that generates a temporary, energy-consuming shield to absorb a limited amount of damage.

I hope you find some of these suggestions useful and can incorporate them in the future. Again, I love the mod! (P.S. Please excuse any errors in my English, as it's not my first language).

**Reply by Gahedes** [reply]
These ideas sound very good

**Comment by PyjamallamaAU** [comment]
Just a question and was not sure where to put it so adding it here.

Oritech has support for Applied Energistics, is there any planned compatiblity and integration for Refined Storage in the same way?

Great mod btw.
